team-validator.js

```diff
@@ -133,7 +140,7 @@ class Validator {
 			set.name = null;
 		}
 		set.species = set.species;
-		set.name = set.name || set.baseSpecies;
+		set.name = set.name || (set.isOtherForm ? set.species : set.baseSpecies);
```

Comment:
For reasons I can't figure out, letting `set.name` be `set.baseSpecies` here results in the server sending a species id rather than something like `Vivillon-Garden`.